### PR TITLE
[15-wires] implement Maller's optimization

### PR DESCRIPTION
**Pull request by mimoo**
**DO NOT MERGE**: I only implemented this for 15-wires, not for 15-wires-plookup. To make tests work I deleted 15-wires-plookup. We are still figuring out how to deal with 15-wires-plookup.

In this PR I implement the Maller's optimization for 15-wires. 
What is Maller's optimization? Or how does the protocol change? I documented it here: https://o1-labs.github.io/mina-book/crypto/plonk/maller_15.html

Some commits are also creating a o1-utils crate to deduplicate code and move extension to the arkwork types to their own crate (and avoid dependency issues).

### CODE REVIEW

File: dlog/plonk-15-wires/src/prover.rs [hunk 1]
```diff
@@ -260,91 +262,74 @@ where
...
-            + &index.cs.endomul_lnrz(&e, &alpha[range::ENDML]))
-            + &index.cs.vbmul_lnrz(&e, &alpha[range::MUL]))
-            + &index.cs.perm_lnrz(&e, &oracles, &alpha[range::PERM]);
-
-        evals[0].f = f.eval(evlp[0], index.max_poly_size);
-        evals[1].f = f.eval(evlp[1], index.max_poly_size);
+        let t_chunked = t.chunk_polynomial(zeta_n, index.max_poly_size);
+        let ft: DensePolynomial<Fr<G>> = &f_chunked - &t_chunked.scale(zeta_n - Fr::<G>::one());
+        let ft_eval0 = ft.evaluate(&zeta);
+        println!("prover's ft_eval0 = {}", ft_eval0);
```
Comment (imeckler) [inline dlog/plonk-15-wires/src/prover.rs:331]: delete?

File: dlog/plonk-15-wires/src/prover.rs [hunk 2]
```diff
@@ -355,17 +340,18 @@ where
         let p_eval = if p.is_zero() {
             [Vec::new(), Vec::new()]
         } else {
-            [vec![p.evaluate(&evlp[0])], vec![p.evaluate(&evlp[1])]]
+            [vec![p.evaluate(&zeta)], vec![p.evaluate(&zeta_omega)]]
```
Comment (imeckler) [inline dlog/plonk-15-wires/src/prover.rs:343]: Nice, these names are much clearer.

**Review by imeckler — APPROVED** [review]
Looks great! Just made one comment

**Comment by mimoo** [comment]
rebased and addressed comments

**Pull request merged**